tests/Persistence/Mapping/ColocatedMappingDriverTest.php

```diff
@@ -48,46 +50,86 @@ public function testAddGetExcludePaths(): void
 
     public function testGetSetFileExtension(): void
     {
-        $driver = $this->createDriver(__DIR__ . '/_files/colocated');
+        $driver = $this->createDirectoryPathDriver(__DIR__ . '/_files/colocated');
         self::assertSame('.php', $driver->getFileExtension());
 
         $driver->setFileExtension('.php1');
 
         self::assertSame('.php1', $driver->getFileExtension());
     }
 
-    /** @dataProvider pathProvider */
-    public function testGetAllClassNames(string $path): void
+    /** @dataProvider directoryPathProvider */
+    public function testGetAllClassNamesForDirectory(string $dirPath): void
     {
-        $driver = $this->createDriver($path);
+        $driver = $this->createDirectoryPathDriver($dirPath);
 
         $classes = $driver->getAllClassNames();
 
         sort($classes);
         self::assertSame([Entity::class, EntityFixture::class], $classes);
     }
 
+    public function testGetAllClassNamesForFilePaths(): void
+    {
+        $driver = $this->createFilePathsDriver([
+            __DIR__ . '/_files/colocated/Entity.php',
+            __DIR__ . '/_files/colocated/TestClass.php',
+        ]);
+
+        $classes = $driver->getAllClassNames();
+
+        self::assertSame([Entity::class], $classes, 'The driver should only return the class names for the provided file path names, excluding transient class names.');
+    }
+
+    public function testGetAllClassNamesWorksBothForFilePathsAndRetroactivelyAddedDirectoryPaths(): void
+    {
+        $driver = $this->createFilePathsDriver([__DIR__ . '/_files/colocated/Entity.php']);
+
+        $driver->addPaths([__DIR__ . '/_files/colocated/']);
+
+        $classes = $driver->getAllClassNames();
+        sort($classes);
+
+        self::assertSame(
+            [Entity::class, EntityFixture::class],
+            $classes,
+            'The driver should return class names from both the provided file path names and the retroactively added directory paths (these should not be ignored).',
+        );
+    }
+
     /** @return Generator<string, array{string}> */
-    public static function pathProvider(): Generator
+    public static function directoryPathProvider(): Generator
     {
         yield 'straigthforward path' => [__DIR__ . '/_files/colocated'];
         yield 'winding path' => [__DIR__ . '/../Mapping/_files/colocated'];
     }
 
-    private function createDriver(string $path): MyDriver
+    private function createDirectoryPathDriver(string $dirPath): MyDriver
     {
-        return new MyDriver([$path]);
+        return new MyDriver([$dirPath]);
+    }
+
+    /** @param list<string> $filePaths */
+    private function createFilePathsDriver(array $filePaths): MyDriver
+    {
+        return new MyDriver(new ArrayIterator($filePaths));
```

Comment:
No, this is too ugly, sorry. You don't want to document something similar in ORM and ODM. The fact that a `symfony/finder` is an `Iterator` is cool, but it shouldn't become a requirement. The ORM and ODM must stay usable without the Symfony bundles.

```suggestion
        return new MyDriver($filePaths);
```

Reply:
Agreeing with you. Your provided solution is neat, so let's stick with it.